pkg/cache/cache.go

```diff
@@ -76,22 +76,27 @@ func init() {
 // Get will return the cached configuration of the given ServiceAccount.
 // It will first look at the set of ServiceAccounts configured using annotations. If none are found, it will look for any
 // ServiceAccount configured through the pod-identity-webhook ConfigMap.
-func (c *serviceAccountCache) Get(name, namespace string) (role, aud string, useRegionalSTS bool, tokenExpiration int64) {
+func (c *serviceAccountCache) Get(name, namespace string) (role, aud string, useRegionalSTS bool, tokenExpiration int64, err error) {
 	klog.V(5).Infof("Fetching sa %s/%s from cache", namespace, name)
+	var respSA *CacheResponse
 	{
-		resp := c.getSA(name, namespace)
-		if resp != nil && resp.RoleARN != "" {
-			return resp.RoleARN, resp.Audience, resp.UseRegionalSTS, resp.TokenExpiration
+		respSA = c.getSA(name, namespace)
+		klog.V(5).Infof("Service account %s/%s not found in cache", namespace, name)
```

Comment:
You don't need to log here. At this point you don't know whether service account is found or not.